### PR TITLE
[issue-650] raise ParsingError if elements don't have an SPDXID

### LINKED ISSUE #650: RDF-parser raises no error if an element has no SPDXID
When trying to parse a file that contains an element (file, package, snippet) without `SPDXID` no error is raised. Instead the element gets a random ID as it is a blank node (e.g. "N5c02cc1dd7c345b7abeaef214e1fa3e8"). The validation catches this as it is no valid SPDXID but I think this should raise an error when parsing.

### PR DESCRIPTION
**Pull request by meretp**
fixes #650

### CODE REVIEW

File: tests/spdx/parser/rdf/test_file_parser.py [hunk 1]
```diff
@@ -35,3 +37,18 @@ def test_parse_file():
...
+
+
+def test_parse_invalid_file():
+    graph = Graph().parse(os.path.join(os.path.dirname(__file__), "data/invalid_documents/file_without_spdx_ids.xml"))
+    file_node = graph.value(predicate=RDF.type, object=SPDX_NAMESPACE.File)
+    doc_namespace = "https://some.namespace"
+
+    assert isinstance(file_node, BNode)
+    with pytest.raises(SPDXParsingError) as err:
+        _ = parse_file(file_node, graph, doc_namespace)
```
Comment (armintaenzertng) [inline tests/spdx/parser/rdf/test_file_parser.py:49]:
```suggestion
        parse_file(file_node, graph, doc_namespace)
```
Also at the other locations.

File: src/spdx_tools/spdx/parser/rdf/graph_parsing_functions.py [hunk 2]
```diff
@@ -140,7 +140,7 @@ def get_value_from_graph(
     # this is a helper method to cast some rdf types from graph.value() to be compatible with the
     # code that follows
     value = graph.value(subject=subject, predicate=predicate, object=_object, default=default, any=_any)
-    if value and not isinstance(value, (URIRef, Literal, BNode)):
+    if value != _any and value and not isinstance(value, (URIRef, Literal, BNode)):
```
Comment (armintaenzertng) [inline src/spdx_tools/spdx/parser/rdf/graph_parsing_functions.py:143]: I don't get the `value != _any` part here. `_any` is a bool, why should it be equal to `value`?

File: src/spdx_tools/spdx/parser/rdf/package_parser.py [hunk 3]
```diff
@@ -108,7 +108,9 @@ def parse_package(package_node: URIRef, graph: Graph, doc_namespace: str) -> Pac
     )
     homepage = parse_literal(logger, graph, package_node, DOAP.homepage)
     attribution_texts = []
-    for _, _, attribution_text_literal in graph.triples((package_node, SPDX_NAMESPACE.attributionText, None)):
+    for _, _, attribution_text_literal in get_correctly_typed_triples(
+        logger, graph, package_node, SPDX_NAMESPACE.attributionText, None
+    ):
```
Comment (armintaenzertng) [inline src/spdx_tools/spdx/parser/rdf/package_parser.py:113]: Is this a separate issue? I don't see the connection to missing SPDX Ids

**Review by armintaenzertng — COMMENTED** [review]
Thanks for the fix, I have a couple of questions, though.

File: src/spdx_tools/spdx/parser/rdf/graph_parsing_functions.py (hunk 2 above)
Comment (meretp) [reply src/spdx_tools/spdx/parser/rdf/graph_parsing_functions.py:143]: Yeah, sorry, mixed the values here. This needs to be `value != default`. While working on this I noticed that an error about wrong types would be logged if a `package` has no `files_analyzed` field. This is as the changes you commented below not related to the origin issue and rather a fix for the wrapper methods introduced with #563. Shall I create a separate PR for these changes?

File: src/spdx_tools/spdx/parser/rdf/package_parser.py (hunk 3 above)
Comment (meretp) [reply src/spdx_tools/spdx/parser/rdf/package_parser.py:113]: Yes, as commented above a separate issue.

File: tests/spdx/parser/rdf/test_file_parser.py (hunk 1 above)
Comment (meretp) [reply tests/spdx/parser/rdf/test_file_parser.py]: fixed

File: src/spdx_tools/spdx/parser/rdf/graph_parsing_functions.py (hunk 2 above)
Comment (armintaenzertng) [reply src/spdx_tools/spdx/parser/rdf/graph_parsing_functions.py:143]: Yes, please do :)

Comment (meretp) [reply src/spdx_tools/spdx/parser/rdf/graph_parsing_functions.py:143]: https://github.com/spdx/tools-python/pull/653 
The PR should be merged first as the test in this PR fails without these changes.

**Review by armintaenzertng — APPROVED** [review]
Thanks! :)

**Pull request merged**